src/main/java/edu/cornell/kfs/coa/batch/dataaccess/impl/CopyLegacyAccountAttachmentsDaoJdbc.java

```diff
@@ -0,0 +1,88 @@
+
+public class CopyLegacyAccountAttachmentsDaoJdbc
+        extends CuSqlQueryPlatformAwareDaoBaseJdbc implements CopyLegacyAccountAttachmentsDao {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private static final int MAX_ERROR_MESSAGE_LENGTH = 1000;
+
+    @Override
+    public List<LegacyAccountAttachment> getLegacyAccountAttachmentsToCopy(
+            final int fetchSize, final int maxRetryCount) {
+        final CuSqlQuery sqlQuery = buildLegacyAccountAttachmentQuery(fetchSize, maxRetryCount);
+        final List<LegacyAccountAttachment> results = queryForValues(sqlQuery, this::mapLegacyAccountAttachment);
+        LOG.info("getLegacyAccountAttachmentsToCopy, Fetched {} attachments for copying", results.size());
+        return results;
+    }
+
+    private CuSqlQuery buildLegacyAccountAttachmentQuery(final int fetchSize, final int maxRetryCount) {
+        return new CuSqlChunk()
+                .append("SELECT * FROM KFS.TEMP_ACCT_ATTACH_FOR_COPYING")
+                .append(" WHERE COPIED_IND = ").appendAsParameter(KRADConstants.NO_INDICATOR_VALUE)
+                .append(" AND RETRY_COUNT < ").appendAsParameter(Types.INTEGER, maxRetryCount)
+                .append(" AND FILE_SYSTEM_FILE_NAME IS NOT NULL ")
+                .append(" ORDER BY COPYING_ACCT_ATTACH_ID")
+                .append(" FETCH FIRST ").appendAsParameter(Types.INTEGER, fetchSize).append(" ROWS ONLY")
+                .toQuery();
+    }
+
+    private LegacyAccountAttachment mapLegacyAccountAttachment(final ResultSet rs, final int rowNum)
+            throws SQLException {
+        final LegacyAccountAttachment attachment = new LegacyAccountAttachment();
+        attachment.setId(rs.getLong("COPYING_ACCT_ATTACH_ID"));
+        attachment.setLegacyAccountCode(rs.getString("ORIGINAL_ACCOUNT_CODE"));
+        attachment.setKfsChartCode(rs.getString("KFS_CHART_CODE"));
+        attachment.setKfsAccountNumber(rs.getString("KFS_ACCOUNT_NBR"));
+        attachment.setFileName(rs.getString("FILE_NAME"));
+        attachment.setAddedBy(rs.getString("ADDED_BY"));
+        attachment.setFileDescription(rs.getString("FILE_DESCRIPTION"));
+        attachment.setFilePath(rs.getString("FILE_SYSTEM_FILE_NAME"));
+        attachment.setRetryCount(rs.getInt("RETRY_COUNT"));
+        attachment.setCopied(rs.getString("COPIED_IND"));
+        attachment.setLatestErrorMessage(rs.getString("LATEST_ERR_MSG"));
+        return attachment;
+    }
+
+    @Override
+    public void markLegacyAccountAttachmentAsCopied(final LegacyAccountAttachment legacyAccountAttachment) {
+        final Long id = legacyAccountAttachment.getId();
+        final CuSqlQuery sqlQuery = new CuSqlChunk()
+                .append("UPDATE KFS.TEMP_ACCT_ATTACH_FOR_COPYING")
+                .append(" SET COPIED_IND = ").appendAsParameter(KRADConstants.YES_INDICATOR_VALUE)
+                .append(" WHERE COPYING_ACCT_ATTACH_ID = ").appendAsParameter(Types.BIGINT, id)
```

Comment:
should this update the LATEST_ERR_MSG to an empty string?

Reply:
I'm fine with having the error message cleared out when the row successfully copies, so I'll adjust the SQL accordingly.

Reply:
The queries will set `COPIED_IND` to `'Y'` when download succeeds, or will increment the `RETRY_COUNT` in the event of a failure (or a repeated failure). Those two columns should be enough to help identify the processing state of a particular attachment.